### PR TITLE
Support TF_USE_LEGACY_KERAS

### PR DESCRIPTION
**Pull request by mattdangerw**
Colab testing out all our cases -> https://colab.research.google.com/gist/mattdangerw/43aba025b2587fe132a7f696f6630664/keras-3-detection.ipynb

This covers the case where tensorflow has been configured with TF_USE_LEGACY_KERAS=True. In this case, we will also load the legacy version of Keras.

### CODE REVIEW

File: keras_nlp/backend/config.py [hunk 1]
```diff
@@ -65,8 +62,20 @@
 if "KERAS_BACKEND" in os.environ and os.environ["KERAS_BACKEND"]:
     _MULTI_BACKEND = True
 
-# If keras is version 3, use multi-backend keras (our only option).
-_IS_KERAS_3 = version.parse(keras.__version__) >= version.parse("3.0.0.dev0")
+
+def detect_keras_3():
```
Comment (mattdangerw) [inline keras_nlp/backend/config.py:66]: rename this to be more accurate

**Comment by divyashreepathihalli** [comment]
@mattdangerw may be this function
```
def keras_3():
    """Check if Keras 3 is installed."""
    return _IS_KERAS_3
    ```
    also would need to check if keras 3 is installed without tensorflow

**Comment by mattdangerw** [comment]
Got a verbal lgtm from @fchollet right now, so will pull this in.

**Pull request merged**